metagraph/src/graph/alignment/aligner_alignment.cpp

```diff
@@ -55,19 +55,21 @@ std::ostream& operator<<(std::ostream& out, const Alignment &alignment) {
 }
 
 void Alignment::append(Alignment&& other) {
-    assert(query_.data() + query_.size() == other.query_.data());
+    assert(query_.data() + query_.size() + other.get_clipping() == other.query_.data());
     assert(orientation_ == other.orientation_);
-    assert(cigar_.empty() || cigar_.back().first != Cigar::CLIPPED);
 
     nodes_.insert(nodes_.end(), other.nodes_.begin(), other.nodes_.end());
     sequence_ += std::move(other.sequence_);
     score_ += other.score_;
 
     cigar_.append(std::move(other.cigar_));
-    query_ = { query_.data(), query_.size() + other.query_.size() };
+    query_ = { query_.data(), other.query_.size() + (other.query_.data() - query_.data()) };
```

Comment:
Are you sure about the second argument? Describe what it represents in a comment.

Reply:
ok. this is one of the strange things about `std::string_view`, it takes a length as the second argument instead of an end pointer (they added the latter in C++20)

[https://en.cppreference.com/w/cpp/string/basic_string_view/basic_string_view](url)

Reply:
yeah, but is that a length though?
`other.query_.size() + (other.query_.data() - query_.data())`

Reply:
I guess a simpler way to write it would be
```suggestion
    query_ = std::string_view(query_.data(), other.query_.end() - query_.begin());
```

before, it was complaining about passing a signed integer for the size, but that'll probably avoided by not initializing with `{}`

Reply:
Ok, looks simpler, it's still not obvious, why can you subtract end of one string from begin of another?
Are they segments of the same string and you want to take their union?

If so, writing this in a comment would help.

Reply:
Essentially, `query_` is a `string_view` of a much longer input sequence.

To avoid confusion, I think I should just rename `query_` to something like `query_view_`

Reply:
I added this as a TODO